### PR TITLE
Remove Unsafe method fixup

### PR DESCRIPTION
**Pull request by tajila**
Remove Unsafe method fixup

Anonclass are now stored in the SCC, as a result we can no longer perform bytecode 
fixups after the romClass is created. As a result, virtual methods that are not stored in
the vtable will now be invoked with INVOKEPRIVATE send target. Invoke private will be
enabled in Java8.

### CODE REVIEW

File: runtime/makelib/targets.mk.linux.inc.ftl [hunk 1]
```diff
@@ -43,7 +43,7 @@ $(UMA_DLLTARGET) : $(UMA_OBJECTS) $(UMA_TARGET_LIBRARIES)
 		$(UMA_DLL_LINK_POSTFLAGS)
 ifdef j9vm_uma_gnuDebugSymbols
 	$(OBJCOPY) --only-keep-debug $(UMA_DLLTARGET) $(UMA_DLLTARGET).dbg
-	$(OBJCOPY) --strip-debug $(UMA_DLLTARGET)
+	$(OBJCOPY) $(UMA_DLLTARGET)
```
Comment (DanHeidinga) [inline runtime/makelib/targets.mk.linux.inc.ftl]: accidentally included?  here and below?

File: runtime/vm/BytecodeInterpreter.hpp [hunk 2]
```diff
@@ -697,12 +697,12 @@ class INTERPRETER_CLASS
 		J9Method *method = NULL;
 		void* const jitReturnAddress = VM_JITInterface::peekJITReturnAddress(_currentThread, _sp);
 		UDATA jitVTableOffset = VM_JITInterface::jitVTableIndex(jitReturnAddress, interfaceVTableIndex);
-#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+#if defined(J9VM_OPT_INVOKEPRIVATE)
 		if (J9_ARE_ANY_BITS_SET(jitVTableOffset, J9_VTABLE_INDEX_DIRECT_METHOD_FLAG)) {
 			/* Nestmates: vtable index is really a J9Method to directly invoke */
```
Comment (DanHeidinga) [inline runtime/vm/BytecodeInterpreter.hpp]:
```suggestion
			/* vtable index is really a J9Method to directly invoke */
```

File: runtime/makelib/targets.mk.linux.inc.ftl (hunk 1 above)
Comment (tajila) [reply runtime/makelib/targets.mk.linux.inc.ftl]: Yup, I forgot to remove the debug commit

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux,zlinux,aix,osx,win jdk8

**Comment by DanHeidinga** [comment]
FYI @andrewcraik I don't expect this will affect the JIT but can you confirm that the invokeprivate implementation in the JIT is enabled for Java 8 as well as for 11+?

**Comment by tajila** [comment]
@DanHeidinga I missed one `#ifdef` in resolveSupport. Ill run a build with the new changes and let you know when its ready.

**Comment by andrewcraik** [comment]
So the relevant changes were linked in https://github.com/eclipse/openj9/issues/2673#issuecomment-422842810. From reading those diffs I don't see any ifdefs or other language level checks so I don't see why the JIT won't do the right thing for Java 8. I'd appreciate at it if @DanHeidinga could also make a quick scan to see if there is anything you see that looks like a version check, but to me this looks all unconditional and enabled everywhere all the time.

**Comment by tajila** [comment]
@DanHeidinga changes are ready

**Comment by DanHeidinga** [comment]
> I'd appreciate at it if @DanHeidinga could also make a quick scan to see if there is anything you see that looks like a version check, but to me this looks all unconditional and enabled everywhere all the time.

Agreed.  I don't see any version checks either

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux,zlinux,aix,osx,win jdk8

**Pull request merged**